### PR TITLE
Use assembly definition files; Get play-mode working

### PR DESCRIPTION
**Pull request by AndrewTHEManeri**
### Purpose of this PR

Enables EditorVR support in play-mode, applies assembly definition files for all submodules.  

### Testing status

That's what we're doing here.  Let's be thorough.

### Technical risk

High - the 'editor only' assumption of EXR is going away and we need to be very vigilant looking for unintended side-effects.

### Comments to reviewers

Let's see what kind of velocity we have with all-eyes on everything.  If we're not making much progress we might do a divide-and-conquer approach instead.

### CODE REVIEW

**Comment by amirebrahimi** [comment]
I'm happy to help with this, too. If there are a lot of changes, then one suggestion is to merge this into a local dev-branch here on this repo and then perform the PR from there. Otherwise, we'll need to make changes in the original branch.

File: Scripts/Utilities/ComponentUtils.cs [hunk 1]
```diff
@@ -31,12 +31,14 @@ public static class ComponentUtils
         public static T GetOrAddIf<T>(GameObject go, bool option) where T : Component
         {
             T component = go.GetComponent<T>();
+#if UNITY_EDITOR
             if (option && component == null)
                 component = Undo.AddComponent<T>(go);
-
+#endif
```
Comment (AndrewTHEManeri) [inline Scripts/Utilities/ComponentUtils.cs]: Add the component without undo if in play mode?

File: Shaders/EVRMeshChain.shader [hunk 2]
```diff
@@ -53,7 +53,9 @@
 		// prevent overlapping lines from adding too much color,
 		// we set the alpha value to one after visiting a pixel.
 		Blend OneMinusDstAlpha DstAlpha, One One
-		BlendOp Add, Max
+		// FIXME: There's a bug where the _SMOOTHNESS_TEXTURE_ALBEDO_CHANNEL_A keyword gets set incorrectly causing the alpha to be incorrect and consequently the
```
Comment (AndrewTHEManeri) [inline Shaders/EVRMeshChain.shader]: Note to self: Look into this

File: Scripts/Utilities/MaterialUtils.cs [hunk 3]
```diff
@@ -20,8 +20,12 @@ static class MaterialUtils
         /// <returns>Cloned material</returns>
         public static UnityMaterial GetMaterialClone(Renderer renderer)
         {
+#if UNITY_EDITOR
             // The following is equivalent to renderer.material, but gets rid of the error messages in edit mode
             return renderer.material = UnityObject.Instantiate(renderer.sharedMaterial);
+#else
+            return renderer.material;
```
Comment (AndrewTHEManeri) [inline Scripts/Utilities/MaterialUtils.cs]: Any reason that instantiating the sharedMaterial as above is not appropriate?

**Comment by AndrewTHEManeri** [comment]
@amirebrahimi I'm getting errors about missing the SteamVR assembly and OVR Assembly.  Did you make those manually or are they in updated libraries somewhere?

**Pull request closed without merging**